examples/solids/index.rst

```diff
@@ -244,12 +216,14 @@ We begin by defining two symmetric tensors in the reference configuration, the r
 and the Green-Lagrange strain tensor
 
 .. math::
+   :label: eq-non-linear-strain
+
    \bm E = \frac 1 2 (\bm C - \bm I_3) = \frac 1 2 \Big( \nabla_X \bm u + (\nabla_X \bm u)^T + (\nabla_X \bm u)^T \nabla_X \bm u \Big),
 
-which converges to the linear strain tensor :math:`\bm \epsilon` in the small-deformation limit.
-Our constitutive models will express :math:`\bm S` as a function of :math:`\bm E`, much like the linear case :math:numref:`linear-stress-strain` expresses :math:`\bm\sigma` in terms of :math:`\bm\epsilon`.
+the latter of which converges to the linear strain tensor :math:`\bm \epsilon` in the small-deformation limit.
+The constitutive models considered, appropriate for large deformations, express :math:`\bm S` as a function of :math:`\bm E`, similar to the linear case, showed in equation  :math:numref:`linear-stress-strain`, which  expresses the relationship between :math:`\bm\sigma` and :math:`\bm\epsilon`.
 
-This is a nonlinear tensor-valued function of tensor-valued input, but an arbitrary choice of such a function will generally not be invariant under orthogonal transformation, thus not admissible because a physical model must not depend on the choice of coordinate system chosen to express it.
+Equation :math:numref:`eq-non-linear-strain` represents a nonlinear tensor-valued function of a tensor-valued input, but an arbitrary choice of such a function will generally not be invariant under orthogonal transformations, thus will not admissible because a physical model must not depend on the choice of coordinate system chosen to express it.
```

Comment:
"This" referred to `S(E)` in the previous sentence, not the definition of `E` (which isn't a choice). I'm editing now so no need for you to fix.

Reply:
Sorry. Yes, I misunderstood. In fact, it's better to avoid "this" when 2 things are introduced in the previous sentence.

Reply:
Thanks; agreed.